.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,52 @@
+name: "Code Scanning"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [main]
+  schedule:
+    # Runs every Monday morning PST
+    - cron: "17 15 * * 1"
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: javascript
+          # If you wish to specify custom queries, you can do so here or in a config file.
+          # By default, queries listed here will override any specified in a config file.
+          # Prefix the list here with "+" to use these queries and those in the config file.
+          # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+      # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+      # If this step fails, then you should remove it and run the build manually (see below)
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v1
```

Comment:
If GitHub handles these versions the same way that they do for actions (one tag for v1.1 and one tag for v1) then we might want to add this to dependabot's ignores

Also, it seems that [codeql-action has a v2](https://github.com/github/codeql-action) according to the tags, but idk if that applies to everything or how that all works - I didn't realize that it was possible to have multiple actions per repo!

Reply:
Ooo good point. I'll see if I can figure that out 😬

Reply:
Did I do this right?

https://github.com/cdr/code-server/blob/0934accc84aba9b50dc82b7ccee53b7bd9971ec6/.github/dependabot.yml#L12

Reply:
I think so? I hope so lol. One can never know with dependabot :joy:

Reply:
Right? We both know this never works out 100% of the time 😂 well, easily reversible if this is wrong!